pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -257,16 +260,16 @@ def start_predicting(self, trainer) -> None:
         xmp.spawn(self.new_process, **self.xmp_spawn_kwargs)
 
     def training_step(self, *args, **kwargs):
-        return self.lightning_module.training_step(*args, **kwargs)
+        return self.model(*args, **kwargs)
```

Comment:
This should be `wrapped_model`.

Reply:
Assigning it while moving the model to device. https://github.com/PyTorchLightning/pytorch-lightning/blob/tpu_spawn_added/pytorch_lightning/plugins/training_type/tpu_spawn.py#L172

`MpModelWrapper` is just using mp lock and returns the model https://github.com/pytorch/xla/blob/master/torch_xla/distributed/xla_multiprocessing.py#L432